pants

```diff
@@ -2,6 +2,39 @@
 # Copyright 2014 Pants project contributors (see CONTRIBUTORS.md).
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
+REPO_ROOT=$(cd $(dirname "${BASH_SOURCE[0]}") && pwd -P)
+
+# TODO(#7152): remove this after the python 3 migration!
+function clean_if_interpreter_changed() {
+  _python_version_to_run="$1"
+
+  if [[ -n "${_recursive_pants_shell_script_invocation:-}" ]]; then
+    return 0
+  fi
+
+  _python_version_file="${REPO_ROOT}/.pants-python-version"
+
+  if [[ -f "$_python_version_file" ]]; then
+    _prev_run_python_version="$(cat "$_python_version_file")"
+    if [[ "$_python_version_to_run" == "$_prev_run_python_version" ]]; then
+      return 0
+    fi
+  else
+    cat >&2 <<EOF
+${_python_version_file} not found. Forcing an initial clean-all...
+EOF
+  fi
+
+  export _recursive_pants_shell_script_invocation='y'
+  cat >&2 <<EOF
+Caches must be cleared before switching pants Python versions in this repo.
```

Comment:
This line suggests to me that you should run `./pants clean-all` manually before trying to change interpreters. You'll note however that even if you do that, this script will still run `clean-all`

(As an aside, it is totally okay IMO that a manual `./pants clean-all` will still result in an automated `./pants clean-all` when changing versions. The whole point of this PR is you should never really be having to run that command manually, unless you're trying to kill the daemon etc.)

Instead, would be good to make this sound more automated. Maybe make the two lines be:
```
Different Python interpreter version detected than the last run.
Clearing the cache and preparing a Python ${_python_version_to_run} run...
```

Reply:
This is great, I'll make this change.

Reply:
Done!